android/titanium/src/java/org/appcelerator/titanium/util/TiAnimationBuilder.java

```diff
@@ -1481,18 +1420,23 @@ private void setAnchor(int width, int height, float thisAnchorX, float thisAncho
 		float pivotX = 0, pivotY = 0;
 
 		if (thisAnchorX != Ti2DMatrix.DEFAULT_ANCHOR_VALUE) {
-			pivotX = width * thisAnchorX;
+			pivotX = (width * thisAnchorX);
 		}
 
 		if (thisAnchorY != Ti2DMatrix.DEFAULT_ANCHOR_VALUE) {
 			pivotY = height * thisAnchorY;
 		}
 
-		if (PRE_HONEYCOMB) {
-			setViewPivot(pivotX, pivotY);
-		} else {
-			setViewPivotHC(pivotX, pivotY);
+		// Passing 0 as both X and Y parameters for a view's pivot point
+		// results in setting the point in the view's center for APIs 16
+		// to 19. This is why we add one for both paramters if this is the
+		// case.
+		if (PRE_LOLLIPOP && thisAnchorX == 0 && thisAnchorY == 0) {
+			pivotX++;
+			pivotY++;
```

Comment:
Pivot is a normalized value between 0.0 and 1.0.

So, if given pivot is (0,0) which is the top-left corner, then adding 1 to both will make it pivot at the bottom-right corner, right?

Perhaps adding epsilon would be better?

Reply:
On our end - yes, it is between 0.0 and 1.0. In the Android code it is passed as pixels.
view.setPivotX() leads to
https://android.googlesource.com/platform/frameworks/base.git/+/master/core/java/android/view/RenderNode.java#632

It is multiplied by the view's dimensions here:
https://github.com/appcelerator/titanium_mobile/blob/master/android/titanium/src/java/org/appcelerator/titanium/util/TiAnimationBuilder.java#L1484
and here:
https://github.com/appcelerator/titanium_mobile/blob/master/android/titanium/src/java/org/appcelerator/titanium/util/TiAnimationBuilder.java#L1488

So in the end it will count as the pixel in position (1,1). 
Increasing them with EPSILON will work, too.

Reply:
Ahh... thanks man. I didn't know that. :)